make/go/dep_protoc_gen_go.mk

```diff
@@ -7,8 +7,8 @@ $(call _assert_var,CACHE_VERSIONS)
 $(call _assert_var,CACHE_BIN)
 
 # Settable
-# https://github.com/protocolbuffers/protobuf-go/releases 20240906 checked 20240916
-PROTOC_GEN_GO_VERSION ?= v1.34.3-0.20240906163944-03df6c145d96
+# https://github.com/protocolbuffers/protobuf-go/releases 20241007 checked 20241007
+PROTOC_GEN_GO_VERSION ?= v1.35.1
```

Comment:
@doriable we should be able to upgrade everywhere else to get off the pseudo-version and on to a proper tag.